packages/driver/src/cypress/log.ts

```diff
@@ -524,36 +537,41 @@ function create (Cypress, cy, state, config) {
     }
   }
 
-  const triggerLog = function (log) {
+  triggerLog (log) {
     log._hasInitiallyLogged = true
 
-    return trigger(log, 'command:log:added')
+    return this.trigger(log, 'command:log:added')
   }
 
-  const addToLogs = function (log) {
+  addToLogs (log) {
     const id = log.get('id')
 
-    logs[id] = true
+    this.logs[id] = true
   }
 
-  const logFn = function (options: any = {}) {
-    if (!_.isObject(options)) {
-      $errUtils.throwErrByPath('log.invalid_argument', { args: { arg: options } })
+  // only fire the log:state:changed event
+  // as fast as every 4ms
+  fireChangeEvent (log) {
+    const triggerStateChanged = () => {
+      return this.trigger(log, 'command:log:changed')
     }
 
-    const log = Log(cy, state, config, options)
+    const debounceFn = _.debounce(triggerStateChanged, 4)
 
-    // add event emitter interface
-    $Events.extend(log)
+    return debounceFn()
+  }
+}
 
-    const triggerStateChanged = () => {
-      return trigger(log, 'command:log:changed')
+export function create (Cypress, cy, state, config) {
```

Comment:
any reason the logFn logic couldn't live in the LogManager ?
```
export function create (Cypress, cy, state, config) {
  const logManager = new LogManager()
  return logManager.createLog(cy, state, config, logManager, options)
}
```

Reply:
I left it there because it was there for years. 

After your comment, I also thought about it and it's Manager's job to create `logFn`. So, I moved it.